From a73020a9c388b4882c384087882a77ab2c7b3729 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Fri, 10 Sep 2021 12:28:31 +0200
Subject: [PATCH] Lib/sysconfig.py: use values from build configuration file

This allows correctly substituting them for target builds using
native python.

Upstream-Status: Inappropriate [oe-core cross builds]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>

---
 Lib/sysconfig.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 93c6f73..a87870f 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -668,6 +668,13 @@ def get_config_vars(*args):
             _CONFIG_VARS['VPATH'] = sys._vpath
         if os.name == 'posix':
             _init_posix(_CONFIG_VARS)
+        _CONFIG_VARS['installed_base'] = _CONFIG_VARS['prefix']
+        _CONFIG_VARS['base'] = _CONFIG_VARS['prefix']
+        _CONFIG_VARS['installed_platbase'] = _CONFIG_VARS['prefix']
+        _CONFIG_VARS['platbase'] = _CONFIG_VARS['prefix']
+        _CONFIG_VARS['platlibdir'] = _CONFIG_VARS['PLATLIBDIR']
+        _INSTALL_SCHEMES['posix_prefix']['include'] = _CONFIG_VARS['INCLUDEPY']
+        _INSTALL_SCHEMES['posix_prefix']['platinclude'] = _CONFIG_VARS['INCLUDEPY']
         if _HAS_USER_BASE:
             # Setting 'userbase' is done below the call to the
             # init function to enable using 'get_config_var' in
